kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/RealRenderContext.kt

```diff
@@ -52,31 +53,37 @@ class RealRenderContext<StateT, OutputT : Any>(
   /** Used to prevent modifications to this object after [buildBehavior] is called. */
   private var frozen = false
 
+  @Suppress("OverridingDeprecatedMember")
+  @UseExperimental(ExperimentalCoroutinesApi::class)
   override fun <EventT : Any> onEvent(handler: (EventT) -> WorkflowAction<StateT, OutputT>):
       EventHandler<EventT> {
     checkNotFrozen()
     return EventHandler { event ->
       // Run the handler synchronously, so we only have to emit the resulting action and don't
       // need the update channel to be generic on each event type.
-      val update = handler(event)
+      val action = handler(event)
 
       // If this returns false, we lost the race with another event being sent.
-      check(nextUpdateFromEvent.complete(update)) {
+      check(nextUpdateFromEvent.complete(action)) {
         "Expected to successfully deliver event. Are you using an old rendering?\n" +
             "\tevent=$event\n" +
-            "\tupdate=$update"
+            "\tattempted action=$action\n" +
+            "\tactual action=${nextUpdateFromEvent.getCompleted()}"
       }
     }
   }
 
+  @UseExperimental(ExperimentalCoroutinesApi::class)
   override fun <A : WorkflowAction<StateT, OutputT>> makeActionSink(): Sink<A> {
     checkNotFrozen()
 
     return object : Sink<A> {
       override fun send(value: A) {
         // If this returns false, we lost the race with another event being sent.
         check(nextUpdateFromEvent.complete(value)) {
-          "Expected to successfully deliver $value. Are you using an old rendering?"
+          "Expected to successfully deliver action. Are you using an old rendering?\n" +
```

Comment:
Is this any clearer?

```
"A rendering can only accept a single action.\n" +
    "\tprocessed action=${...}\n"
    "\tlate action=$value" +
```

Reply:
I like the processed/late terminology more.